x-pack/plugins/cases/public/components/all_cases/use_all_cases_state.tsx

```diff
@@ -80,6 +80,47 @@ const validateQueryParams = (queryParams: QueryParams): QueryParams => {
   return { ...queryParams, perPage, sortOrder };
 };
 
+/**
+ * Parses filter options from the URL.
+ * If a filter option is defined as an array in DEFAULT_FILTER_OPTIONS, this function
+ * ensures it's returned as an array even if URLSearchParams provides it as a single value.
+ * URLSearchParams.getAll() returns an empty array if the param is not defined in the URL.
+ * URLSearchParams.get() returns null if the param is not defined in the URL and in case of
+ * multiple values, it returns the first value only
+ */
+const parseURLWithFilterOptions = (search: string) => {
```

Comment:
What about:

```
const parseURLWithFilterOptions = (search: string) => {
  const urlParams = new URLSearchParams(search);

  const parsedUrlParams: { [key: string]: string[] | string | null } = {};

  urlParams.forEach((_, key) => {
    const typedKey = key as keyof FilterOptions;
    const isArray =
      DEFAULT_FILTER_OPTIONS[typedKey] != null && Array.isArray(DEFAULT_FILTER_OPTIONS[typedKey]);

    const value = isArray ? urlParams.getAll(key) : urlParams.get(key);
    parsedUrlParams[key] = value;
  });

  return parsedUrlParams;
};

```

Reply:
Could you please add tests for the new logic?

Reply:
I did update it with something different before I saw your comment, please check it out and let me know. I think/hope 🤞  it's very simple now